tests/apps/qchem/test_dynamics.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Tests for strawberryfields.apps.qchem.dynamics
+"""
+import numpy as np
+import pytest
+
+import strawberryfields as sf
+from strawberryfields.apps.qchem import dynamics
+
+pytestmark = pytest.mark.apps
+
+
+t1 = 10.0
+t2 = 100.0
+a = 1 / np.sqrt(2)
+U1 = np.array([[a, -a], [a, a]])
+U2 = np.array([[a, -a, 0.0], [a, a, 0.0], [0.0, 0.0, 1.0]])
+w1 = np.array([3914.92, 3787.59])
+w2 = np.array([3914.92, 3787.59, 1627.01])
+p1 = np.array(
+    [
+        [0.00000000e00, 0.00000000e00, 2.04851621e-04, 0.00000000e00],
+        [0.00000000e00, 2.82155738e-02, 0.00000000e00, 0.00000000e00],
+        [9.71579575e-01, 0.00000000e00, 0.00000000e00, 0.00000000e00],
+        [0.00000000e00, 0.00000000e00, 0.00000000e00, 0.00000000e00],
+    ]
+)
+p2 = np.array(
+    [
+        [
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.75372632, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+        ],
+        [
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.22889562, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+        ],
+        [
+            [0.01737806, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+        ],
+        [
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+            [0.00000000, 0.00000000, 0.00000000, 0.00000000],
+        ],
+    ]
+)
+
+
+@pytest.mark.parametrize("time, unitary, frequency, prob", [(t1, U1, w1, p1), (t2, U2, w2, p2)])
+@pytest.mark.filterwarnings("ignore: The circuit consists of 2 disconnected components")
```

Comment:
Note that I think this occurs due to the second unitary where mode 3 is independent of modes 1 and 2. You can see this by looking at the last block of `p2` - i.e., a photon from the first mode can't interact with the final mode.

Unless it's a big undertaking, I'd recommend trying a full 3-mode unitary (and removing this warning).

Reply:
Done.